src/app/fdctl/config.c

```diff
@@ -591,6 +591,50 @@ username_to_uid( char * username ) {
   FD_LOG_ERR(( "configuration file wants firedancer to run as user `%s` but it does not exist", username ));
 }
 
+static void
+validate_ports( config_t * result ) {
+ulong solana_port_min, solana_port_max;
+  char dynamic_port_range[ 32 ];
+  fd_memcpy( dynamic_port_range, result->dynamic_port_range, sizeof(dynamic_port_range) );
+
+  char * dash = strstr( dynamic_port_range, "-" );
+  if( FD_UNLIKELY( !dash ) )
+    FD_LOG_ERR(( "configuration specifies invalid [dynamic_port_range] `%s`. "
+                 "This must be formatted like `<min>-<max>`",
+                 result->dynamic_port_range ));
+
+  *dash = '\0';
+  char * endptr;
+  solana_port_min = strtoul( result->dynamic_port_range, &endptr, 10 );
+  if( FD_UNLIKELY( *endptr != '\0' || solana_port_min > USHRT_MAX ) )
+    FD_LOG_ERR(( "configuration specifies invalid [dynamic_port_range] `%s`. "
+                 "This must be formatted like `<min>-<max>`",
+                 result->dynamic_port_range ));
+  solana_port_max = strtoul( dash + 1, &endptr, 10 );
+  if( FD_UNLIKELY( *endptr != '\0' || solana_port_max > USHRT_MAX ) )
```

Comment:
```suggestion
  if( FD_UNLIKELY( *endptr != '\0' || solana_port_max > USHORT_MAX ) )
```
These are firedancer-specific, but more consistent